specification/app/resource-manager/Microsoft.App/preview/2023-04-01-preview/CommonDefinitions.json

```diff
@@ -0,0 +1,1196 @@
+              "enum": [
+                "CNAME",
+                "HTTP",
+                "TXT"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "ManagedCertificateDomainControlValidation",
+                "modelAsString": true
+              }
+            },
+            "validationToken": {
+              "description": "A TXT token used for DNS TXT domain control validation when issuing this type of managed certificates.",
+              "type": "string",
+              "readOnly": true
+            }
+          }
+        }
+      },
+      "x-ms-client-flatten": true
+    },
+    "ManagedCertificatePatch": {
+      "description": "A managed certificate to update",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Application-specific metadata in the form of key-value pairs."
+        }
+      }
+    },
+    "ManagedCertificateCollection": {
+      "description": "Collection of Managed Certificates.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedCertificate"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "JobTemplate": {
+      "description": "Container Apps Job versioned application definition. Defines the desired state of an immutable revision. Any changes to this section Will result in a new revision being created",
+      "type": "object",
+      "properties": {
+        "initContainers": {
+          "description": "List of specialized containers that run before app containers.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/InitContainer"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "containers": {
+          "description": "List of container definitions for the Container App.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Container"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "volumes": {
+          "description": "List of volume definitions for the Container App.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Volume"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        }
+      }
+    },
+    "ServiceBind": {
+      "description": "Configuration to bind a ContainerApp to a dev ContainerApp Service",
+      "type": "object",
+      "properties": {
+        "serviceId": {
```

Comment:
If it will always be an Azure Resource Id, consider using [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details).